core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java

```
@@ -429,6 +432,11 @@ public Response processUpload(
                                                      Text.escapeIllegalJcrChars(driverName),
                                                      Text.escapeIllegalJcrChars(currentFolder));
         fileName = Text.escapeIllegalJcrChars(fileName);
+        if (existenceAction != null && !existenceAction.equals(KEEP) && currentFolderNode.hasNode(fileName)) {
+          if (currentFolderNode.getNode(fileName).isLocked()) {
+            return Response.serverError().entity(NODE_LOCKED).build();
```

According to the eXo REST guidelines, we should return a 403 error : https://community.exoplatform.com/portal/intranet/wiki/group/spaces/platform_4/REST_CRUD_APIs#HResponseCodes
And the content of the response must be a JSON object with the error message, as described in https://community.exoplatform.com/portal/intranet/wiki/group/spaces/platform_4/REST_CRUD_APIs#HClientErrors284xx29. For example, in our case, the developerMessage can contain an error code which will be used by the caller (javascript).